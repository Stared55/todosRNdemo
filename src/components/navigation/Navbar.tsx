import * as React from 'react';
import { SafeAreaView, StatusBar, StyleSheet, Text, View } from 'react-native';
import { Colors } from 'src/utils/styles/colors';
import { yScale } from 'src/utils/styles/scale';

export interface NavbarProps {
  title: string;
}

const Navbar: React.SFC<NavbarProps> = ({ title }) => {
  return (
    <>
      <SafeAreaView style={styles.safeArea} />
      <StatusBar backgroundColor={Colors.white} translucent={false} />
      <View style={styles.navbar}>
        <Text style={styles.label}>{title}</Text>
      </View>
    </>
  );
};
export default Navbar;

const styles = StyleSheet.create({
  safeArea: {
    backgroundColor: Colors.white,
  },
  navbar: {
    justifyContent: 'space-around',
    alignItems: 'center',
    height: yScale(50),
    backgroundColor: Colors.white,
    flexDirection: 'row',
    borderColor: Colors.grey,
    borderBottomWidth: 0.6,
  },
  label: {
    fontSize: yScale(24),
  },
});
