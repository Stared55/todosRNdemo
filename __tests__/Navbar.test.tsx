import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Navbar from 'src/components/navigation/Navbar';
import { Text, SafeAreaView, StatusBar, View } from 'react-native';

describe('Navbar', () => {
  let component: ShallowWrapper;
  let props: any;
  const title = 'Home';
  let children = (
    <>
      <SafeAreaView />
      <StatusBar />
      <View>
        <Text>{title}</Text>
      </View>
    </>
  );

  beforeEach(() => {
    component = shallow(<Navbar title={title}>{children}</Navbar>);
    props = component.props();
    jest.clearAllMocks();
  });

  it('should match to snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it(`should render correct navbar title`, () => {
    const c = component.find({ children: title });
    expect(c).toHaveLength(1);
  });

  it(`should render 3 children`, () => {
    expect(props.children).toHaveLength(3);
  });
});
