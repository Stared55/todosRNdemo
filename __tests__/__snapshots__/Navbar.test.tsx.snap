// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should match to snapshot 1`] = `
<Fragment>
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "backgroundColor": "#ffffff",
      }
    }
  />
  <StatusBar
    animated={false}
    backgroundColor="#ffffff"
    showHideTransition="fade"
    translucent={false}
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderBottomWidth": 0.6,
        "borderColor": "#A19F9D",
        "flexDirection": "row",
        "height": 98.08823529411764,
        "justifyContent": "space-around",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 47.08235294117647,
        }
      }
    >
      Home
    </Text>
  </View>
</Fragment>
`;
